### DISCUSSION TITLE
how to read garden variety PC 360k floppies?

### DISCUSSION THREAD

**Discussion by eyaleb** [keirf/greaseweazle#367] [General]
I can read the 1.2Mb ones just fine, but trying to read the 360k ones fails. I have no trouble reading with `dd if=/dev/fd0 ...`.
Format ibm.360 produces many errors and garbage data.
ibm.scan produces many warnings but does extract the correct data.
Is there a cleaner way to do this?

I saw https://github.com/keirf/greaseweazle/discussions/180#top but trying it produces screens-full of errors and it progresses VERY slowly. It retrieved only 2%. Surely there is a simpler way for such a common format?

GW V4 with fw 1.4, installed package is version 1.15.1.

TIA

**Comment by keirf** [comment]
Try adding option `--dd=L` to `gw read`. The drive probably needs to be informed of density via pin 2. This option will set pin 2 low (by default the pin is left high).

If dumping raw you may also want to try adding `--adjust-speed=300` as the HD drive spins at 360rpm, but DD is normally at 300rpm. However most Greaseweazle `--format=` specifiers will auto-adjust for this.

**Reply by eyaleb** [reply]
This is also what the above linked discussion says, which I already tried to no avail.
I now tried reading in 4 ways

```
--format=ibm.360
--format=ibm.360 --dd=L
--format=ibm.360 --dd=H
--format=ibm.360 --adjust-speed=300
```

Got identical results. gw says it only found 18 sectors. Comparing to the good `dd`(*1) I see that the first 18 sectors are identical (so correct) but the rest is not. So gw reads the first track (both sides, 18 sectors) well but not further.

I then tried a second drive (Mitsumi D509V3) with similar failure. This drive has jumpers that can be adjusted (e.g. https://forum.vcfed.org/index.php?threads/floppy-drive-jumper-settings.56473/) but it works OK as is (with simple `dd`).

I kept the logs from the runs if it helps.
BTW, in case this matters, it should be clear that I am using Linux (hence `dd`). I do not have a Windows machine to test.

(*1) The disk contains a large zip file and the good `dd` copy tests and extracts OK so it is a valid reference.

**Reply by keirf** [reply]
An excerpt of the logs would have been helpful. I'm now pretty sure it must be full of "Ignoring unexpected sector" messages.

Probably you need to specify double-stepping as it's a 40t disk in an 80t drive. So add `--tracks=step=2` to your command line. You may also still need `--dd=L`, you'll have to experiment.

**Reply by eyaleb** [reply]
Yes! using `--format=ibm.360 --tracks=step=2` works, with both my drive and my friend's.
Adding `--dd=L` was not necessary but worked just as well.

Thanks.
